app/controllers/work_package_types/types_controller.rb

```diff
@@ -0,0 +1,179 @@
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module WorkPackageTypes
+  class TypesController < ApplicationController
+    include PaginationHelper
+
+    layout "admin"
+
+    before_action :require_admin
+    before_action :find_type, only: %i[update move destroy]
+
+    current_menu_item do
+      :types
+    end
+
+    def index
+      @types = ::Type
+                .includes(:workflows, :projects, :custom_fields, :color)
+                .page(page_param)
+                .per_page(per_page_param)
+    end
+
+    def type
+      @type
+    end
+
+    def new
+      @type = Type.new(params[:type])
+      load_projects_and_types
+    end
+
+    def edit
+      if params[:tab].blank?
+        redirect_to tab: :settings
+      else
+        type = ::Type.includes(:projects, :custom_fields).find(params[:id])
+        render_edit_tab(type)
+      end
+    end
+
+    def create
+      additional_params = {}
+      value = params.dig(:type, :copy_workflow_from)
+      additional_params[:copy_workflow_from] = value if value.present?
+
+      service_call = WorkPackageTypes::CreateService
+                      .new(user: current_user)
+                      .call(permitted_type_params.merge(additional_params))
+
+      @type = service_call.result
+      if service_call.success?
+        redirect_to edit_type_settings_path(@type), notice: t(:notice_successful_create), status: :see_other
+      else
+        render action: :new, status: :unprocessable_entity
+      end
+    end
+
+    def update
+      UpdateTypeService
+        .new(@type, current_user)
+        .call(permitted_type_params) do |call|
+        call.on_success { redirect_to_type_tab_path(@type, t(:notice_successful_update)) }
+        call.on_failure { render_edit_tab(@type, status: :unprocessable_entity) }
+      end
+    end
```

Comment:
Is this still called somewhere?

Reply:
Good catch, thanks! I think, `update` is not needed here anymore